contracts/peg/NonCustodialPSM.sol

```diff
@@ -0,0 +1,460 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {Decimal} from "../external/Decimal.sol";
+import {Constants} from "../Constants.sol";
+import {OracleRef} from "./../refs/OracleRef.sol";
+import {RateLimited} from "./../utils/RateLimited.sol";
+import {INonCustodialPSM} from "./INonCustodialPSM.sol";
+import {IPCVDeposit, PCVDeposit} from "./../pcv/PCVDeposit.sol";
+import {GlobalRateLimitedMinter} from "./../utils/GlobalRateLimitedMinter.sol";
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+/// @notice Peg Stability Module that holds no funds.
+/// On a mint, it transfers all proceeds to a PCV Deposit and then deposits them into the deposit's target
+/// When funds are needed for a redemption, they are simply pulled from the PCV Deposit
+contract NonCustodialPSM is
+    INonCustodialPSM,
+    RateLimited,
+    OracleRef,
+    PCVDeposit,
```

Comment:
imo it shouldn't be a PCVDeposit. Assume it holds no assets

Reply:
done